eclair-core/src/main/scala/fr/acinq/eclair/router/RouteCalculation.scala

```diff
@@ -382,7 +394,11 @@ object RouteCalculation {
   /** Update used capacity by taking into account an HTLC sent to the given route. */
   private def updateUsedCapacity(route: Route, usedCapacity: mutable.Map[ShortChannelId, MilliSatoshi]): Unit = {
     route.hops.reverse.foldLeft(route.amount) { case (amount, hop) =>
-      usedCapacity.updateWith(hop.shortChannelId)(previous => Some(amount + previous.getOrElse(0 msat)))
+      val id = hop match {
+        case hop: ChannelHop => hop.shortChannelId
+        case hop: BlindedHop => hop.dummyId
```

Comment:
Why not just put `shortChannelId` in the `ConnectedHop` trait?

Reply:
Because that leads to bugs! This isn't a real `localAlias`, and by putting it in `ConnectedHop` and not thinking about it, we accidentally pass it on to components that expect an scid that matches one of our channels. I found one such case in one of your earlier PR, so I decided that we should make sure it's very explicit when we use that `dummyId`, to help us catch these issues.